### PR TITLE
feat: ✨ add unlock with password, add gs output handling

### PR DESCRIPTION
**Pull request by callegarimattia**
Add password field in load api to unlock protected pdf. Parse ghostscript output and retrieve result of the processing

### CODE REVIEW

**Comment by callegarimattia** [comment]
> Please update the docs as well
I'll do it as soon as I get home. Windows not happy to use `make`

File: tests/test_load.py [hunk 1]
```diff
@@ -298,3 +324,21 @@ def test_span_has_no_characters(valid_file_name):
...
+def test_extract_spans_text(valid_file_name):
+    hot_pdf_object = HotPdf()
+    hot_pdf_object.load(valid_file_name)
+    text = hot_pdf_object.extract_spans_text(0, 0, 1000, 1000)
+
+    assert len(text) == 608 if os.name == "nt" else 728
+
+
+@pytest.mark.skip("FAILING")
+def test_CONSISTENCY(valid_file_name):
```
Comment (krishnasism) [inline tests/test_load.py:338]: lol

File: tests/test_load.py [hunk 2]
```diff
@@ -28,11 +29,35 @@ def non_existent_file_name():
...
+
+
 def test_load(valid_file_name):
     hot_pdf_object = HotPdf()
     hot_pdf_object.load(valid_file_name)
 
 
+def test_load_locked(locked_file_name):
+    hot_pdf_object = HotPdf()
+    hot_pdf_object.load(locked_file_name, password="hotpdfiscool")
```
Comment (krishnasism) [inline tests/test_load.py:44]: agree

File: hotpdf/processor.py [hunk 3]
```diff
@@ -1,38 +1,78 @@
...
 
     Args:
         file_path (str): The path to the PDF file.
+        password (str): The password to use to unlock the file
         first_page (int): The first page to extract.
         last_page (int): The last page to extract.
-
+    Raises:
+        PermissionError: If the password is missing or wrong
     Returns:
```
Comment (krishnasism) [inline hotpdf/processor.py:27]: RuntimeError is missing in docstring

**Review by krishnasism — CHANGES_REQUESTED** [review]

File: hotpdf/processor.py (hunk 3 above)
Comment (callegarimattia) [reply hotpdf/processor.py:27]: good catch

File: hotpdf/processor.py [hunk 4]
```diff
@@ -1,38 +1,78 @@
...
+        )
+        status = validate_gs_output(output)
+    except subprocess.CalledProcessError as err:
+        status = Result.UNKNOWN_ERROR
+        logging.error(err)
+
+    return status
+
+
+def clean_xlm(temporary_xml_file_name: str) -> None:
```
Comment (krishnasism) [inline hotpdf/processor.py:67]: clean_xml instead of clean_xlm

**Review by krishnasism — CHANGES_REQUESTED** [review]

File: hotpdf/processor.py (hunk 4 above)
Comment (callegarimattia) [reply hotpdf/processor.py:67]: 💀

**Review by krishnasism — CHANGES_REQUESTED** [review]

**Review by krishnasism — APPROVED** [review]

**Comment by krishnasism** [comment]
lgtm

File: hotpdf/hotpdf.py [hunk 5]
```diff
@@ -61,16 +62,19 @@ def load(
 
         Args:
             pdf_file (str): The path to the PDF file to be loaded.
+            password (str): Password to use to unlock the pdf
```
Comment (callegarimattia) [inline hotpdf/hotpdf.py:65]:
```suggestion
            password (str, optional): Password to use to unlock the pdf
```

**Review by callegarimattia — COMMENTED** [review]
Fixed 👍🏻

**Pull request merged**